flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/SpillableSubpartitionView.java

```diff
@@ -108,11 +108,7 @@ int releaseMemory() throws IOException {
 				for (int i = 0; i < numBuffers; i++) {
 					Buffer buffer = buffers.remove();
 					spilledBytes += buffer.getSize();
-					try {
-						spillWriter.writeBlock(buffer);
-					} finally {
-						buffer.recycle();
-					}
+					spillWriter.writeBlock(buffer);
```

Comment:
Here is the place where you depend on recycling which is not guaranteed by `spillWriter`'s interface.

Reply:
Actually, if I see this correctly, here the original code is wrong since it is already recycling a buffer which was added to an asynchronous file write operation. This would lead to data corruption if the buffer is re-used in the meanwhile, wouldn't it?!